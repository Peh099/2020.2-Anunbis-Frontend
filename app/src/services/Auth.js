import api from './Api'

export function isAuthenticated(){
  return getToken() !== undefined;
}

export function isProfessor(){
  if(getToken()){
    return localStorage.getItem('professor') ? true : false;
  }
}

export function isStudent(){
  if(getToken()){
    return localStorage.getItem('student') ? true : false;
  }
}

export function getToken() {
  return (localStorage.getItem('access_token'));
}

export async function sendLogin(email, password, callback, errorCallback) {
  const body = {
    email: email,
    password: password
  }

  function writeUser(data){
    localStorage.setItem('access_token', data.access_token);
    if (data.user.reg_student)
      localStorage.setItem('student', JSON.stringify(data.user));
    else 
      localStorage.setItem('professor', JSON.stringify(data.user));
  }

  api.post("/login", body)
    .then(response => {
      const data = response.data;
      if (data.access_token)
        writeUser(data);
      else
        logOut();
    })
    .then(() => callback())
    .catch(() => errorCallback())
}

export function logOut() {
  localStorage.removeItem('access_token');
  localStorage.removeItem('student');
  localStorage.removeItem('professor');
}