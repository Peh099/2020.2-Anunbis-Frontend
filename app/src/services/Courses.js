import api from './Api';

export const getCourses = (setCourses) => {
  api.get('/course').then((response) => {
    console.log(response);
    setCourses(response.data);
  });
};
