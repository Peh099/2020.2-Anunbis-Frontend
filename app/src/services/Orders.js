export function orderPop(post1, post2) {
  if (post1?.feedbacks.agrees > post2?.feedbacks.agrees) return -1;
  if (post1?.feedbacks.agrees < post2?.feedbacks.agrees) return 1;
  return 0;
}

export function orderDate(post1, post2) {
  if (post1?.post_date > post2?.post_date) return -1;
  if (post1?.post_date < post2?.post_date) return 1;
  return 0;
}

export function orderHRate(post1, post2) {
  if (post1?.rating > post2?.rating) return -1;
  if (post1?.rating < post2?.rating) return 1;
  return 0;
}

export function orderLRate(post1, post2) {
  if (post1?.rating < post2?.rating) return -1;
  if (post1?.rating > post2?.rating) return 1;
  return 0;
}

export const orders = [
  { fun: orderPop, id: 0, name: 'Mais popular', selected: true },
  { fun: orderDate, id: 1, name: 'Mais recente' },
  { fun: orderHRate, id: 2, name: 'Maior Nota Geral' },
  { fun: orderLRate, id: 3, name: 'Menor Nota Geral' },
];
