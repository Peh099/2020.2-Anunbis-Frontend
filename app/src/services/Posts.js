import api from './Api';

export const getPosts = (setPosts) => {
  api.get('post').then((response) => {
    if (response.status === 200) setPosts(response.data);
  });
};
