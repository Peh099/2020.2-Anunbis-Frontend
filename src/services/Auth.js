import api from './Api'

export function getToken() {
  return localStorage.getItem('access_token');
}

export async function sendLogin(email, password, callback) {
  const body = {
    email: email,
    password: password
  }
  
  api.post("/login", body)
    .then(response => {
      const data = response.data;
      if (data.access_token) {
        localStorage.setItem('access_token', data.access_token);
        localStorage.setItem('student', JSON.stringify(data.student));
      }
      else
        localStorage.removeItem('access_token', 'student');
    })
    .then(() => callback())
}

export function logOut() {
  localStorage.removeItem('access_token', 'student');
}

export default function isAuthenticated() {
  return (getToken() != null) ? true : false;
};

