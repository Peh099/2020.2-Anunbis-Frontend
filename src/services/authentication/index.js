import React from 'react';

const isAuthenticated = localStorage.getItem('acessToken') ? true : false;


export async function sendLogin(url, body){
    fetch(url, {
        method: 'post',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify(body)
      })
      .then(response => {
        let data = response;
        if (data.token){
            localStorage.setItem('acessToken', data.token);
        }
      })
}

export default isAuthenticated;