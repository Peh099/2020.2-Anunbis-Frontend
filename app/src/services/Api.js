import axios from 'axios';
import { getToken } from './Auth';

const api = axios.create({
  baseURL: process.env.REACT_APP_API_HOST,
});

api.interceptors.request.use(
  (config) => {
    const accessToken = getToken();
    config.headers.Authorization = `Bearer ${accessToken}`;
    return config;
  },
  (error) => {
    console.log(error);
  },
);

export default api;
