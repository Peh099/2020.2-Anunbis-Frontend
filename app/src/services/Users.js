import { getToken } from './Auth';

const createUser = (localStorageName, homePath) => {
    return {
        localStorageName: localStorageName,
        homePath: homePath,
        isAuthenticated: () => getToken() ? localStorage.getItem(localStorageName) ? true : false : false
    }
}

const Users = {
    PROFESSOR: createUser("professor", '/professor/'),
    STUDENT: createUser('student', '/student/'),
    VISITANT: createUser('visitant', '/'),
    isAuthenticated: () => getToken() !== null,
    whoAuthenticated: function () {
        if (getToken()) {
            const userName = Object.keys(this).find((user) => 
                localStorage.getItem(this[user]?.localStorageName) !== null
            )
            return this[userName];
        }
        return this.VISITANT;
    }
}

export default Users;