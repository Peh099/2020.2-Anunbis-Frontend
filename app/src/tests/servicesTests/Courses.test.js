import { getCourses } from '../../services/Courses';

describe('Snapshot Auth component', () => {
  test('must call the callback on getting Courses', (done) => {
    const json = {
      id: 1,
      name: 'Engenharia de Software',
    };
    function callback(data) {
      expect(data).toStrictEqual([json]);
      done();
    }
    getCourses(callback);
  });
});
