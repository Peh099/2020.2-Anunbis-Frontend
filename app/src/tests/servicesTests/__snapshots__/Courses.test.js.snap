// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Auth component matches the snapshot 1`] = `null`;
